src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -54,6 +54,15 @@ public SqlConnection(string connectionString) : this()
             CacheConnectionStringProperties();
         }
 
+        private SqlConnection(SqlConnection connection)
+        {
+            GC.SuppressFinalize(this);
```

Comment:
I don't have a definitive answer. I see that the GC.SuppressFinalize() call in the first line of constructor of many System.Data related classes in Framework. However in the port to .Net core, many of these calls were removed. E.g. In `public SqlConnection() ` these calls are present in the Framework. 
https://github.com/Microsoft/referencesource/blob/master/System.Data/System/Data/ProviderBase/DbConnectionHelper.cs#L36 

To try to find the answers, I tried to dig into the history of the framework source code to see if I can find any clues, but there is nothing specific about this particular LoC.
I suspect that this may have something to do with the Native SNI objects being held by the SqlConnection and not wanting to release those unmanaged objects during garbage collection.